src/core/Compare.h

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright (C) 2018 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef KEEPASSXC_COMPARE_H
+#define KEEPASSXC_COMPARE_H
+
+#include <QDateTime>
+
+#include "core/Clock.h"
+
+enum CompareOption
```

Comment:
For now I don't have any better name for CompareOption. For now it is only a way to collect all possible options for comparisons to prevent scattering and duplicating the logic all over the classes. This way, the client is responsible to select the usable and appropriate members but has consistent behavior and api for comparisons. For now I'm out of ideas for better naming or a better implementation strategy, but I'm open for proposals and improvements in this regard.

Reply:
The members of this enum are quite specific to database items. Perhaps make that part of the name.